### PR TITLE
Add process to take care of gene counts

### LINKED ISSUE #37: Add process to generate gene counts 4 drop

### PR DESCRIPTION
**Pull request by projectoriented**
Closes #37

### CODE REVIEW

**Comment by projectoriented** [comment]
Why it shouldn't require baseDir implicit var: https://www.nextflow.io/docs/latest/faq.html#how-do-i-invoke-custom-scripts-and-tools

File: modules/main.nf [hunk 1]
```diff
@@ -209,6 +209,23 @@ process index_bam{
     """
 }
 
+process generate_gene_counts4drop{
+
+	input:
+		tuple val(sample), path(counts)
+
+	output:
+		path('external_geneCounts.tsv'), emit: processed_gene_counts
```
Comment (jemten) [inline modules/main.nf:218]: How would this work with a multisample run. Don't you need to either add sample? Like:
`tuple val(sample), path('external_geneCounts.tsv'), emit: processed_gene_counts`

File: modules/main.nf [hunk 2]
```diff
@@ -209,6 +209,23 @@ process index_bam{
     """
 }
 
+process generate_gene_counts4drop{
+
+	input:
+		tuple val(sample), path(counts)
```
Comment (jemten) [inline modules/main.nf:215]: Wouldn't you need an optional input with external_geneCounts.tsv in order to provide the background database?

**Review by jemten — COMMENTED** [review]
Thanks for the PR Mei!
Some questions regarding the generate_gene_counts4drop process

File: modules/main.nf (hunk 1 above)
Comment (projectoriented) [reply modules/main.nf:218]: Nope! Here it's just one file we need from this process and its next one (sample annot process) parses the contents of the file for sample IDs. If in the case we don't have gene counts for a sample, then we'll take the tuple from output of STAR 😸

File: modules/main.nf (hunk 2 above)
Comment (projectoriented) [reply modules/main.nf:215]: A good question. In that scenario I'll need to write a function in the script to merge different external_geneCounts.tsv files

Comment (projectoriented) [reply modules/main.nf]: No, not here. I was thinking gene counts for background db should be a separate tsv file

Comment (jemten) [reply modules/main.nf]: Ah OK

File: bin/generate_gene_counts.py [hunk 3]
```diff
@@ -114,19 +116,34 @@ def file_exists(path_tsv):
...
+    # file_exist = file_exists("external_geneCounts.tsv")
+    # in_dict = read_star_gene_cnts(args.sample, args.star, args.strandedness)
+    # if file_exist:
+    #     read_existing_gene_cnts(file_exist)
+    #     transform_to_table(
+    #         pad_table(in_dict, read_existing_gene_cnts(file_exist)),
+    #         args.output,
+    #     )
+    # else:
+    #     transform_to_table(in_dict, args.output)
```
Comment (jemten) [inline bin/generate_gene_counts.py:138]: Should we remove this part?

**Review by jemten — APPROVED** [review]

File: bin/generate_gene_counts.py (hunk 3 above)
Comment (projectoriented) [reply bin/generate_gene_counts.py]: Done!

**Pull request merged**